course/api/views.py

```diff
@@ -257,7 +257,12 @@ def __module_to_dict(self, module, **kwargs):
     def list(self, request, *args, **kwargs):
         modules = []
         for module in self.content.data['modules']:
-            modules.append(self.__module_to_dict(module, **kwargs))
+            if module['status'] in (
+                CourseModule.STATUS.READY,
+                CourseModule.STATUS.UNLISTED,
+                CourseModule.STATUS.MAINTENANCE,
+            ):
```

Comment:
```suggestion
            if module['status'] != CourseModule.STATUS.HIDDEN:
```
This feels simpler and appropriate in this case.